# Performance metrics on classification problems
For better understanding when to use true positive/true negative  

In [1]:
from sklearn import metrics

In [2]:
# consider 3 cattegories:
C="Cat"
D='Dog'
F="Fox"

In [3]:
# true values
y_true=[C,C,C,C,C,C ,F,F,F,F,F,F,F,F,F,F, D,D,D,D,D,D,D,D,D]
# predicted values
y_pred=[C,C,C,C,D,F ,C,C,C,C,C,C,D,D,F,F, C,C,C,D,D,D,D,D,D]
# print the confusion matrix
print(metrics.confusion_matrix(y_true,y_pred))

#print he precision and recall among other metrics
print(metrics.classification_report(y_true, y_pred, digits=3))

[[4 1 1]
 [3 6 0]
 [6 2 2]]
              precision    recall  f1-score   support

         Cat      0.308     0.667     0.421         6
         Dog      0.667     0.667     0.667         9
         Fox      0.667     0.200     0.308        10

    accuracy                          0.480        25
   macro avg      0.547     0.511     0.465        25
weighted avg      0.581     0.480     0.464        25



If we look at the matrix above, the accuracy is given by the diagonal terms multiplied minus the others.  
The precision for Cat is the correctly predicted Cat from all predicted Cat. In the confussion matrix, the first column is all the cats predictions, so the accuracy is 4/(4+3+6)=0.308  
Dog=0.667  
Fox=0.667
